com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferWriter.cs

```diff
@@ -0,0 +1,1042 @@
+                return;
+            }
+            if (value is NetworkObject)
+            {
+                WriteValueSafe((NetworkObject)value);
+                return;
+            }
+            if (value is NetworkBehaviour)
+            {
+                WriteValueSafe((NetworkBehaviour)value);
+                return;
+            }
+            if (value is INetworkSerializable)
+            {
+                //TODO ((INetworkSerializable)value).NetworkSerialize(new NetworkSerializer(this));
+                return;
+            }
+
+            throw new ArgumentException($"{nameof(FastBufferWriter)} cannot write type {value.GetType().Name} - it does not implement {nameof(INetworkSerializable)}");
+        }
+
+        /// <summary>
+        /// Write an INetworkSerializable
+        /// </summary>
+        /// <param name="value">The value to write</param>
+        /// <typeparam name="T"></typeparam>
+        public void WriteNetworkSerializable<T>(in T value) where T : INetworkSerializable
+        {
+            // TODO
+        }
+
+        /// <summary>
+        /// Get the required amount of space to write a GameObject
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static int GetWriteSize(GameObject value)
+        {
+            return sizeof(ulong);
+        }
+
+        /// <summary>
+        /// Get the required amount of space to write a GameObject
+        /// </summary>
+        /// <returns></returns>
+        public static int GetGameObjectWriteSize()
+        {
+            return sizeof(ulong);
+        }
+
+        /// <summary>
+        /// Write a GameObject
+        /// </summary>
+        /// <param name="value">The value to write</param>
+        public void WriteValue(GameObject value)
+        {
+            var networkObject = (value).GetComponent<NetworkObject>();
+            if (networkObject == null)
+            {
+                throw new ArgumentException($"{nameof(FastBufferWriter)} cannot write {nameof(GameObject)} types that does not has a {nameof(NetworkObject)} component attached. {nameof(GameObject)}: {(value).name}");
+            }
+
+            if (!networkObject.IsSpawned)
+            {
+                throw new ArgumentException($"{nameof(FastBufferWriter)} cannot write {nameof(NetworkObject)} types that are not spawned. {nameof(GameObject)}: {(value).name}");
+            }
+
+            WriteValue(networkObject.NetworkObjectId);
+        }
+
+        /// <summary>
+        /// Write a GameObject
+        ///
+        /// "Safe" version - automatically performs bounds checking. Less efficient than bounds checking
+        /// for multiple writes at once by calling VerifyCanWrite.
+        /// </summary>
+        /// <param name="value">The value to write</param>
+        public void WriteValueSafe(GameObject value)
+        {
+            var networkObject = (value).GetComponent<NetworkObject>();
```

Comment:
Hey @ShadauxCat I suggest converting this GetComponent call to TryGetComponent, as the latter does not allocate anything (in Editor)